Content.Server/Power/Events/PowerPulseEvent.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace Content.Server.Power.Events;
+
+/// <summary>
+///     Invoked on a target entity, when it was pulsed with an energy.
+///     For instance, interacted with an active stun baton.
+/// </summary>
+public sealed class PowerPulseEvent : EventArgs
+{
+    public EntityUid? User;
+    public EntityUid? Used;
+}
```

Comment:
out of scope (will probably do this myself later) but combining this and `BeingMicrowavedEvent` into a common EMP event would be nice

Reply:
Yeah, maybe made EMP event with power variable. So microwave or stun baton just cause little EMP pulse, not enough to fry electronics on airlocks or APC.